File: sleepy-test/java-junit/AsyncRequestQueueEq_LogEvent_OutOfMemoryException_AsyncNotCrash.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class AsyncRequestQueueEq {

    @Test
    public void logEvent_OutOfMemoryException_AsyncNotCrash() throws Exception {
        try (NoThrowNLogExceptions noThrow = new NoThrowNLogExceptions()) {
            LogFactory logFactory = new LogFactory().setup().loadConfiguration(builder -> {
                builder.forLogger().writeTo(new NoMemoryTarget() {{ setName("LowMem"); }}).withAsync();
            }).getLogFactory();
            Logger logger = logFactory.getCurrentClassLogger();
            NoMemoryTarget target = logFactory.getConfiguration().findTargetByName("LowMem");

            logger.info("Testing1");
            logFactory.flush();
            assertEquals(1, target.getFailedCount());

            logger.info("Testing2");
            for (int i = 0; i < 5000; ++i) {
                if (target.getWriteCount() != 1) break;
                Thread.sleep(1);
            }
            assertEquals(2, target.getFailedCount());

            target.setLowMemory(false);
            logger.info("Testing3");
            for (int i = 0; i < 5000; ++i) {
                if (target.getWriteCount() != 2) break;
                Thread.sleep(1);
            }
            assertEquals(2, target.getFailedCount());
            assertEquals(3, target.getWriteCount());
        }
    }
}

File: refactored/duplicate-assert/CSharpXUnit/CSharpXUnit.Tests/Tests/AsyncRequestQueueEq_LogEvent_OutOfMemoryException_AsyncNotCrash.cs
using Xunit;

public class AsyncRequestQueueEq
{
    [Fact]
    public void LogEvent_OutOfMemoryException_AsyncNotCrash1()
    {
        using (var noThrowNLogExceptions = new NoThrowNLogExceptions())
        {
            var logFactory = new LogFactory().Setup().LoadConfiguration(builder =>
            {
                builder.ForLogger().WriteTo(new NoMemoryTarget() { Name = "LowMem" }).WithAsync();
            }).GetLogFactory();
            var logger = logFactory.GetCurrentClassLogger();
            var target = logFactory.GetConfiguration().FindTargetByName("LowMem");
            logger.Info("Testing1");
            logFactory.Flush();
            Assert.Equal(1, target.GetFailedCount());
            logger.Info("Testing2");
            for (int i = 0; i < 5000; ++i)
            {
                if (target.GetWriteCount() != 1)
                    break;
                System.Threading.Thread.Sleep(1);
            }

            Assert.Equal(2, target.GetFailedCount());
        }
    }

    [Fact]
    public void LogEvent_OutOfMemoryException_AsyncNotCrash2()
    {
        using (var noThrowNLogExceptions = new NoThrowNLogExceptions())
        {
            var logFactory = new LogFactory().Setup().LoadConfiguration(builder =>
            {
                builder.ForLogger().WriteTo(new NoMemoryTarget() { Name = "LowMem" }).WithAsync();
            }).GetLogFactory();
            var logger = logFactory.GetCurrentClassLogger();
            var target = logFactory.GetConfiguration().FindTargetByName("LowMem");
            logger.Info("Testing1");
            logFactory.Flush();
            Assert.Equal(1, target.GetFailedCount());
            logger.Info("Testing2");
            for (int i = 0; i < 5000; ++i)
            {
                if (target.GetWriteCount() != 1)
                    break;
                System.Threading.Thread.Sleep(1);
            }

            target.SetLowMemory(false);
            logger.Info("Testing3");
            for (int i = 0; i < 5000; ++i)
            {
                if (target.GetWriteCount() != 2)
                    break;
                System.Threading.Thread.Sleep(1);
            }

            Assert.Equal(2, target.GetFailedCount());
            Assert.Equal(3, target.GetWriteCount());
        }
    }
}
